app/scripts/metamask-controller.js

```diff
@@ -2527,40 +2737,51 @@ export default class MetamaskController extends EventEmitter {
         registerOnboarding: this.onboardingController.registerOnboarding,
       }),
     );
+    engine.push(this.permissionLogController.createMiddleware());
```

Comment:
I noticed that previously this middleware was only added for sites, not for UI connections. But I guess we'd want our log to include permission changes made within our UI as well :thinking:

Reply:
Hm, yeah, I can't recall why we didn't do it in the first place, but I don't see why not! (It's also possible we were doing it in some convoluted way in the original `PermissionsController`, but w/e.)